apollo-federation/src/sources/connect/models/validation.rs

```diff
@@ -514,6 +569,169 @@ fn parse_url(value: &Node<Value>, coordinate: &str, sources: &SourceMap) -> Resu
     Ok(url)
 }
 
+fn validate_header_arg<'a>(
+    directive_name: &Name,
+    argument_name: &str,
+    headers: &'a Node<Value>,
+    source_map: &SourceMap,
+    object: Option<&Name>,
+    field: Option<&Name>,
+) -> Result<&'a Node<Value>, Vec<Message>> {
+    let mut errors = Vec::new();
+
+    headers
+        .as_list()
+        .map(|l| l.iter().filter_map(|o| o.as_object()).collect_vec())
```

Comment:
I don't _think_ you need to `collect_vec()` here

Reply:
I tried removing it but it caused errors